lib/svcd/svcd.lua

```diff
@@ -101,22 +101,38 @@ SVCD.wcdispatch = function(pay, srcip, srcport)
     end
 end
 
-SVCD.write = function (targetip, svcid, attrid, payload, timeout_ms, on_done)
-    local ivkid = SVCD.ivkid
-    SVCD.ivkid = SVCD.ivkid + 1
-    if SVCD.ivkid > 65535 then
-        SVCD.ivkid = 0
-    end
-    SVCD.handlers[ivkid] = on_done
+SVCD.write = storm.n.svcd_write
+-- SVCD.write = function (targetip, svcid, attrid, payload, timeout_ms, on_done)
+--     print("Nonsexy write")
+--     local ivkid = SVCD.ivkid
+
+--     SVCD.ivkid = SVCD.ivkid + 1
+--     if SVCD.ivkid > 65535 then
+--         SVCD.ivkid = 0
+--     end
+
+--     SVCD.handlers[ivkid] = on_done
+   
+--     storm.os.invokeLater(timeout_ms*storm.os.MILLISECOND, function()
+--         if SVCD.handlers[ivkid] ~= nil then
+--             SVCD.handlers[ivkid](SVCD.TIMEOUT)
+--             SVCD.handlers[ivkid] = nil
+--         end
+--     end)
+--     storm.net.sendto(SVCD.wcsock, storm.mp.pack({svcid, attrid, ivkid, payload}), targetip, 2526)
+-- end
+
+-- svcd_write c impl. temporary function -- working on anonymous function calling
+SVCD.write_invoke_later = function(ivkid, timeout_ms)
     storm.os.invokeLater(timeout_ms*storm.os.MILLISECOND, function()
         if SVCD.handlers[ivkid] ~= nil then
             SVCD.handlers[ivkid](SVCD.TIMEOUT)
             SVCD.handlers[ivkid] = nil
         end
     end)
-    storm.net.sendto(SVCD.wcsock, storm.mp.pack({svcid, attrid, ivkid, payload}), targetip, 2526)
 end
```

Comment:
Yea, we still require the invoke later. We were trying to switch it over to C and we got stuck.